Build failure: python3Packages.ansible-runner

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

While bumping `ansible-core` in https://github.com/NixOS/nixpkgs/pull/435368 we have stumbled across a build failure for `ansible-runner` (and `ansible-kernel` that invokes `ansible-runner`).
The failures are in the tests, can be reproduced with running `nix-build -A python312Packages.ansible-runner` on current master (b2c8bfb1300ebd50bac41fba58e90c62f92a14b7).
hydra seems to build it happily. Why is that?

### Can Hydra reproduce this build failure?

No, Hydra cannot reproduce this build failure.

### Link to Hydra build job

_No response_

### Relevant log output

```console
=================================== FAILURES ===================================
___________________ test_output_when_given_invalid_playbook ____________________
[gw1] linux -- Python 3.12.11 /nix/store/18l19gnl8dq75v6knghbcgkrikmpc0yv-python3-3.12.11/bin/python3.12

tmp_path = PosixPath('/build/pytest-of-nixbld/pytest-0/popen-gw1/test_output_when_given_invalid0')

    def test_output_when_given_invalid_playbook(tmp_path):
        # As shown in the following issue:
        #
        #   https://github.com/ansible/ansible-runner/issues/29
        #
        # There was a lack of output by runner when a playbook that doesn't exist
        # is provided.  This was fixed in this PR:
        #
        #   https://github.com/ansible/ansible-runner/pull/34
        #
        # But no test validated it.  This does that.
        private_data_dir = str(tmp_path)
        ex = init_runner(
            private_data_dir=private_data_dir,
            inventory='localhost ansible_connection=local ansible_python_interpreter="{{ ansible_playbook_python }}"',
            envvars={"ANSIBLE_DEPRECATION_WARNINGS": "False"},
            playbook=os.path.join(private_data_dir, 'fake_playbook.yml')
        )

        ex.run()
        with ex.stdout as f:
            stdout = f.read()
>       assert "ERROR! the playbook:" in stdout
E       AssertionError: assert 'ERROR! the playbook:' in '\x1b[0;31m[ERROR]: the playbook: /build/pytest-of-nixbld/pytest-0/popen-gw1/test_output_when_given_invalid0/fake_playbook.yml could not be found\x1b[0m\n'

test/integration/test_display_callback.py:368: AssertionError
----------------------------- Captured stdout call -----------------------------
[ERROR]: the playbook: /build/pytest-of-nixbld/pytest-0/popen-gw1/test_output_when_given_invalid0/fake_playbook.yml could not be found
___________________________ test_include_role_events ___________________________
[gw1] linux -- Python 3.12.11 /nix/store/18l19gnl8dq75v6knghbcgkrikmpc0yv-python3-3.12.11/bin/python3.12

project_fixtures = PosixPath('/build/pytest-of-nixbld/pytest-0/popen-gw1/test_include_role_events0/projects')

    def test_include_role_events(project_fixtures):
        r = run(
            private_data_dir=str(project_fixtures / 'use_role'),
            playbook='use_role.yml'
        )
        role_events = [event for event in r.events if event.get('event_data', {}).get('role', '') == "benthomasson.hello_role"]
        assert 'runner_on_ok' in [event['event'] for event in role_events]
        for event in role_events:
            event_data = event['event_data']
            assert not event_data.get('warning', False)  # role use should not contain warnings
            assert 'resolved_role' not in event_data  # should not specify FQCN name if not from collection
            if event['event'] == 'runner_on_ok':
                assert event_data['res']['msg'] == 'Hello world!'
            if event['event'] == 'playbook_on_task_start':
>               assert event_data['resolved_action'] == 'ansible.builtin.debug'
E               AssertionError: assert None == 'ansible.builtin.debug'

test/integration/test_events.py:142: AssertionError
```

### Additional context

_No response_

### System metadata

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.41, NixOS, 25.11 (Xantusia), 25.11.20250809.85dbfc7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.4`
 - nixpkgs: `/nix/store/c39q7pww80997lz0fyqxpvsr29q6f7ld-source`

```

### Notify maintainers

@GaetanLepage 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/ansible-runner/default.nix
+++ b/pkgs/development/python-modules/ansible-runner/default.nix
@@ -17,6 +17,7 @@
   importlib-metadata,
 
   # tests
+  addBinToPathHook,
   ansible-core,
   glibcLocales,
   mock,
@@ -26,6 +27,7 @@
   pytest-xdist,
   pytestCheckHook,
   versionCheckHook,
+  writableTmpDirAsHomeHook,
 }:
 
 buildPythonPackage rec {
@@ -60,6 +62,7 @@ buildPythonPackage rec {
   ++ lib.optionals (pythonOlder "3.10") [ importlib-metadata ];
 
   nativeCheckInputs = [
+    addBinToPathHook
     ansible-core # required to place ansible CLI onto the PATH in tests
     glibcLocales
     mock
@@ -69,12 +72,11 @@ buildPythonPackage rec {
     pytest-xdist
     pytestCheckHook
     versionCheckHook
+    writableTmpDirAsHomeHook
   ];
   versionCheckProgramArg = "--version";
 
   preCheck = ''
-    export HOME=$(mktemp -d)
-    export PATH="$PATH:$out/bin";
     # avoid coverage flags
     rm pytest.ini
   '';
@@ -90,8 +92,10 @@ buildPythonPackage rec {
     # Assertion error
     "test_callback_plugin_censoring_does_not_overwrite"
     "test_get_role_list"
+    "test_include_role_events"
     "test_include_role_from_collection_events"
     "test_module_level_no_log"
+    "test_output_when_given_invalid_playbook"
     "test_resolved_actions"
   ];
 

